src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -60,23 +87,47 @@ public DriveSubsystem() {
     m_rightFollowerMotor.follow(m_rightLeaderMotor);
 
     m_drive = new DifferentialDrive(m_leftLeaderMotor, m_rightLeaderMotor);
+    
+    // This code limits the motor current to 40 amps to prevent overheating
+
+    int rpm=3000;
+    int peakCurrent=70;
+    int continuousCurrent=40;
+
+    m_leftLeaderMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_rightLeaderMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_leftFollowerMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+    m_rightFollowerMotor.setSmartCurrentLimit(peakCurrent,continuousCurrent,rpm);
+
 
   }
 
   public void curvatureDrive(double moveSpeed, double rotateSpeed) {
     m_drive.curvatureDrive(moveSpeed, rotateSpeed, true);
   }
 
+  
+  public void pidControlObjects() {
+    //Sets the speed of motors based on the PID control 
+
+    m_leftLeaderMotor.set(leftSpeedPID.getOutputMax());
+    m_rightLeaderMotor.set(rightSpeedPID.getOutputMax());
+    m_leftFollowerMotor.set(leftSpeedPID.getOutputMax());
+    m_rightFollowerMotor.set(rightSpeedPID.getOutputMax());
+  }
+  //it controls the range of output of the PID controller
+  public void pidOutputRange(){
```

Comment:
This method isn't being called from the constructor (for configuration initialization). Though I don't know what our range values should be yet, or if it's necessary.